docs/api/ethereum.md

```diff
@@ -1,5 +1,645 @@
+| options.networkName | A `string` representing which ethereum network we will connect to.
+| options.requiredConfirmations | An `integer` representing the number of confirmations needed for mutations to be considered confirmed. It defaults to `1`.
+| options.signMethod | An `integer` representing the signature type. The available options are `0` (eth_sign) or `2` (EIP-712) or `3` (perosnal_sign). It defaults to `0`.
+| options.unsafeRecipientIds | A list of `strings` representing smart contract addresses that do not support safe ERC-721 transfers (e.g. CryptoKitties address should be listed here). 
+| options.valueLedgerSource | A `string` representing the URL to the compiled ERC-20 related smart contract definition file. This file is used when deploying new value ledgers to the network.
+
+**Usage**
+
+```ts
+import { BitskiProvider } from '@0xcert/ethereum-bitski-backend-provider';
+
+const provider = new BitskiProvider({
+    clientId: '',
+    credentialsId: '',
+    credentialsSecret: '',
+});
+```
+
+**See also:**
+
+[Bitski front-end provider](#bitski-front-end-provider)
+
+### accountId
+
+A class instance `variable` holding a `string` which represents user's current Ethereum wallet address.
+
+### assetLedgerSource
+
+A class instance `variable` holding a `string` which represents the URL to the compiled ERC-721 related smart contract definition file. This file is used when deploying new asset ledgers to the network.
+
+### emit(event, ...options);
+
+A `synchronous` class instance `function` to manually trigger a provider event.
+
+**Arguments:**
+
+| Argument | Description
+|-|-
+| event | [required] A `string` representing a [provider event](./ethereum.md#provider-events) name.
+| options | [required] Pass valid current and previous account ID for `ACCOUNT_CHANGE` event or valid current and previous network version for `NETWORK_CHANGE` event.
+
+**Result:**
+
+An instance of the same provider class.
+
+**Example:**
+
+```ts
+import { AccountEvent } from '@0xcert/ethereum-bitski-backend-provider';
+
+mutation.emit(AccountEvent.NETWORK_CHANGE, '3');
+```
+
+### getAvailableAccounts()
+
+An `asynchronous` class instance `function` which returns currently available Ethereum wallet addresses.
+
+**Result:**
+
+A list of `strings` representing Ethereum account IDs.
+
+**Example:**
+
+```ts
+// perform query
+const accountIds = await provider.getAvailableAccounts();
+```
+
+### getInstance(options)
+
+A static class `function` that returns a new instance of the BitskiProvider class (alias for `new BitskiProvider`).
+
+**Arguments**
+
+See the class [constructor](#bitski-provider) for details.
+
+**Usage**
+
+```ts
+import { BitskiProvider } from '@0xcert/ethereum-bitski-backend-provider';
+
+// create provider instance
+const provider = BitskiProvider.getInstance();
+```
+
+### getNetworkVersion()
+
+An `asynchronous` class instance `function` which returns Ethereum network version (e.g. `1` for Ethereum Mainnet).
```

Comment:
`An `asynchronous` class instance `function` which returns the Ethereum network version (e.g., `1` for Ethereum Mainnet).`